eips_code/eip-3540.py

```diff
@@ -6,85 +6,92 @@
 S_DATA = 0x02
 
 
-def validate_eof(code: bytes):
-    # Old-style contracts are still allowed
-    if len(code) == 0 or code[0] != FORMAT:
-        return
+# Determines if code is in EOF format of any version.
+def is_eof(code: bytes) -> bool:
+    return len(code) >= 2 and code[0] == FORMAT and code[1] == MAGIC
+
 
-    # Validate format and magic
-    assert(len(code) >= 3 and code[1] == MAGIC and code[2] == VERSION)
+# Validate EOF code.
+def validate_eof(code: bytes):
+    # Check version
+    assert len(code) >= 3 and code[2] == VERSION
```

Comment:
I think this still needs to check the magic?

Reply:
No. This function requires EOF as input therefore the MAGIC is already checked. This can be expressed as `assert is_eof(code)` but we also used `assert` for any other check therefore this wouldn't be clear.